odxtools/utils.py

```diff
@@ -34,7 +35,7 @@ def dataclass_fields_asdict(obj: Any) -> Dict[str, Any]:
     This makes hierarchies of dataclasses possible while initializing
     the base class using common code.
     """
-    return {x.name: getattr(obj, x.name) for x in obj.__dataclass_fields__.values()}
+    return {x: getattr(obj, x) for x in signature(type(obj)).parameters}
```

Comment:
inspect is usually slow, are you sure there are no performance side effects?

Reply:
Maybe use `fields` ? https://docs.python.org/3/library/dataclasses.html#dataclasses.fields

Reply:
good point. I got a 30% performance regression when listing one of my bigger files (which is dominated by loading the database). Using your suggestion leaves peformance virtually unchanged. Thanks for the proposal, fixed: 54f928968ea3e76c2ef30d594c9dcbbd33b29cd6 .

(in this commit I also extended the docstring with the reason why we need `dataclass_fields_asdict()` in the first place...)